TCutility/analysis/ts_vibration.py

```diff
@@ -0,0 +1,107 @@
+
+    outputmol = data.molecule.output
+
+    base = np.array(outputmol).reshape(-1,3)
+    tsimode = np.array(data.properties.vibrations.modes[0]).reshape(-1,3)
+
+    posvib = outputmol.copy()
+    posvib.from_array(base + tsimode)
+    posvib.guess_bonds(dmax=bond_tolerance)
+    negvib = outputmol.copy()
+    negvib.from_array(base - tsimode)
+    negvib.guess_bonds(dmax=bond_tolerance)
+
+    pairs = np.where(posvib.bond_matrix() - negvib.bond_matrix())
+    rc = np.array(
+                [   [a+1, b+1, np.sign(posvib.bond_matrix()[a][b] - negvib.bond_matrix()[a][b])]
+                    for a, b in np.c_[pairs[0], pairs[1]] 
+                    if a < b and avg_relative_bond_length_delta(outputmol, posvib, negvib, a, b) > min_delta_dist
+                ],  dtype=int)
+    return rc
+
+
+def validate_transitionstate(calc_dir: str, rcatoms = None, bond_tolerance: float = 1.28, min_delta_dist: float = 0.1) -> bool:
+    '''Function to determine whether a transition state calculation yielded the expected transition state.
+
+    Args:
+        calc_dir: path pointing to the desired calculation.
+        rcatoms: list or array containing expected reaction coordinates, to check against the transition state. If not defined, it is obtained from the ams.rkf user input
+        bond_tolerance: parameter for plams.molecule.guess_bonds() function, used in determine_ts_reactioncoordinate
+        min_delta_dist: minimum relative bond length change before qualifying as active atom, used in determine_ts_reactioncoordinate
+
+    Returns:
+        Boolean value, True if the found transition state reaction coordinates match the expected reaction coordinates, False otherwise
+    '''
+
+    result = determine_ts_reactioncoordinate(calc_dir, bond_tolerance, min_delta_dist)
+
+    if isinstance(rcatoms, list):
+        rcatoms = np.array(rcatoms)
+
+    if not isinstance(rcatoms, np.ndarray):
+        data = read(calc_dir)
+        assert 'reactioncoordinate' in data.input.transitionstatesearch, 'Reaction coordinate is a required input, but was neither provided nor present in the .rkf file'
+        rcatoms = np.array([[int(float(y)) for y in x.split()[1:]] for x in data.input.transitionstatesearch.reactioncoordinate])
+
+    if not isinstance(rcatoms[0], np.ndarray):
+        rcatoms = np.array([rcatoms])
+
+    assert np.all([len(x)==3 for x in rcatoms]), 'Invalid format of reaction coordinate'
+
+    # sort for consistency
+    for index, [a,b,c] in enumerate(rcatoms):
+        if a > b:
+            rcatoms[index] = [b,a,c]
+    rcatoms = rcatoms[rcatoms[:,1].argsort()]
+    rcatoms = rcatoms[rcatoms[:,0].argsort()]
+    result = result[result[:,1].argsort()]
+    result = result[result[:,0].argsort()]
+
+    # only the first reaction coordinate sign is arbitrary, check remaining coordinates for consistency
+    result[:, 2] = result[:, 2] if np.sign(result[0][2]) == rcatoms[0][2] else -result[:, 2]
```

Comment:
Only check if the sign is correct if it is given in rcatoms in the arguments. Otherwise skip it